packages/core/src/index.ts

```diff
@@ -780,13 +780,8 @@ class Hopr extends EventEmitter {
   public async ping(destination: PeerId): Promise<{ info?: string; latency: number }> {
     let start = Date.now()
 
-    let pingResult: HeartbeatPingResult
-    try {
-      pingResult = await this.heartbeat.pingNode(destination)
-    } catch (err) {
-      log(`Could not ping ${destination.toString()}.`, err)
-      return { latency: -1, info: 'error' }
-    }
+    // Propagate any errors thrown upwards
+    let pingResult = await this.heartbeat.pingNode(destination)
```

Comment:
So this will lead to a `HTTP 422` error when called through the API, which makes sense.

Reply:
The API docs already document the 422 error, see https://github.com/hoprnet/hoprnet/blob/1b0460f00a60785c38c61e47d64bc5a7f8ab4f26/packages/hoprd/src/api/v2/paths/node/ping.ts#L112

So I added some additional info to that error.